File: MPP/MPP Proiect/CSharp/protobuf/ProtoUtils.cs
using System.Collections.Generic;
using Java;

namespace protobuf{
    public class ProtoUtils{
        public static Request CreateLoginRequest(model.Referee referee){
            var myReferee = new Referee{
                Id = referee.id,
                Name = referee.Name,
                Email = referee.Email,
                Password = referee.Password
            };
            return new Request{
                Type = Request.Types.Type.Login,
                Referee = myReferee
            };
        }

        public static Request CreateLogoutRequest(model.Referee referee){
            var myReferee = new Referee{
                Id = referee.id,
                Name = referee.Name,
                Email = referee.Email,
                Password = referee.Password
            };
            return new Request{
                Type = Request.Types.Type.Logout,
                Referee = myReferee
            };
        }

        public static Request CreateAddTestRequest(model.Test test){
            var myTest = new Test{
                IdParticipant = test.idParticipant,
                IdReferee = test.idReferee,
                Points = test.points,
                Date = test.date
            };
            return new Request{
                Type = Request.Types.Type.AddTest,
                Test = myTest
            };
        }

        public static Request CreateAllParticipantsRequest(){
            return new Request{
                Type = Request.Types.Type.GetAllParticipants
            };
        }

        public static Request CreateAllActiveParticipantsRequest(int refereeId){
            return new Request{
                Type = Request.Types.Type.GetAllActiveParticipants,
                Id = refereeId
            };
        }

        public static model.Test GetTest(Response response){
            var test = response.Test;
            return new model.Test(test.IdParticipant, test.IdReferee, test.Points,
                test.Date){
                id = test.Id
            };
        }

        public static model.Referee GetReferee(Response response){
            var referee = response.Referee;
            return new model.Referee(referee.Name, referee.Email, referee.Password){
                id = referee.Id
            };
        }

        public static string GetError(Response response){ return response.Error; }

        public static IEnumerable<model.Participant>
            GetParticipants(Response response){
            List<model.Participant> participants = new List<model.Participant>();
            foreach(var participant in response.Participants){
                model.Participant currentParticipant = new model.Participant(
                    participant.Name,
                    participant.TotalPoints){
                    id = participant.Id
                };
                participants.Add(currentParticipant);
            }

            return participants;
        }

        public static Response CreateLoginResponse(model.Referee referee){
            var myReferee = new Referee{
                Name = referee.Name,
                Email = referee.Email,
                Password = referee.Password
            };
            myReferee.Id = referee.id;
            return new Response{
                Type = Response.Types.Type.Ok,
                Referee = myReferee
            };
        }

        public static Response CreateErrorResponse(string eMessage){
            return new Response{
                Type = Response.Types.Type.Error,
                Error = eMessage
            };
        }

        public static Response CreateOkResponse(){
            return new Response{
                Type = Response.Types.Type.Ok
            };
        }

        public static Response CreateGetAllParticipantsResponse(
            IEnumerable<model.Participant> getAllParticipantsSortedByName){
            var participants = new List<Participant>();
            foreach(var participant in getAllParticipantsSortedByName){
                var myParticipant = new Participant{
                    Id = participant.id,
                    Name = participant.Name,
                    TotalPoints = participant.TotalPoints
                };
                participants.Add(myParticipant);
            }
            return new Response{
                Type = Response.Types.Type.GetAllParticipants,
                Participants ={participants}
            };
        }

        public static Response CreateGetAllActiveParticipantsResponse(
            IEnumerable<model.Participant> getAllActiveParticipants){
            var participants = new List<Participant>();
            foreach(var participant in getAllActiveParticipants){
                var myParticipant = new Participant{
                    Id = participant.id,
                    Name = participant.Name,
                    TotalPoints = participant.TotalPoints
                };
                participants.Add(myParticipant);
            }
            return new Response{
                Type = Response.Types.Type.GetAllActiveParticipants,
                Participants ={participants}
            };
        }


        public static model.Referee GetReferee(Request request){
            var referee = request.Referee;
            return new model.Referee(referee.Name, referee.Email, referee.Password){
                id = referee.Id
            };
        }

        public static model.Test GetTest(Request request){
            var test = request.Test;
            return new model.Test(test.IdParticipant, test.IdReferee, test.Points,
                test.Date){
                id = test.Id
            };
        }

        public static int GetId(Request request){
            return request.Id;
        }

        public static Response CreateTestAddedResponse(model.Test test){
            var myTest = new Test{
                IdParticipant = test.idParticipant,
                IdReferee = test.idReferee,
                Points = test.points,
                Date = test.date,
                Id = test.id
            };
            return new Response{
                Type = Response.Types.Type.TestAdded,
                Test = myTest
            };
        }

       
    }
}

File: MPP/MPP Proiect/Java/Networking/src/main/java/protobuffprotocol/ProtoUtils.java
package protobuffprotocol;

import model.Participant;
import model.Referee;
import model.Test;

import java.util.ArrayList;

public class ProtoUtils{
    public static Test getTest(Protobufs.Response updateResponse){
        Test test = new Test();
        test.setIdParticipant(updateResponse.getTest().getIdParticipant());
        test.setIdReferee(updateResponse.getTest().getIdReferee());
        test.setPoints(updateResponse.getTest().getPoints());
        test.setDate(updateResponse.getTest().getDate());
        test.setId(updateResponse.getTest().getId());
        return test;
    }

    public static Referee getReferee(Protobufs.Response response){
        Referee referee = new Referee();
        referee.setName(response.getReferee().getName());
        referee.setEmail(response.getReferee().getEmail());
        referee.setPassword(response.getReferee().getPassword());
        referee.setId(response.getReferee().getId());
        return referee;
    }

    public static String getError(Protobufs.Response response){
        String error = response.getError();
        return error;
    }

    public static Iterable<Participant> getAllParticipants(Protobufs.Response response){
        ArrayList<Participant> participants = new ArrayList<>();
        for(Protobufs.Participant participant : response.getParticipantsList()){
            Participant participant1 = new Participant(participant.getName(),
                    participant.getTotalPoints());
            participant1.setId(participant.getId());
            participants.add(participant1);
        }
        return participants;
    }


    public static Referee getReferee(Protobufs.Request request){
        Referee referee = new Referee(request.getReferee().getName(),
                request.getReferee().getEmail(), request.getReferee().getPassword());
        referee.setId(request.getReferee().getId());
        return referee;
    }

    public static Test getTest(Protobufs.Request request){
        Test test = new Test();
        test.setIdParticipant(request.getTest().getIdParticipant());
        test.setIdReferee(request.getTest().getIdReferee());
        test.setPoints(request.getTest().getPoints());
        test.setDate(request.getTest().getDate());
        test.setId(request.getTest().getId());
        return test;
    }


    public static int getId(Protobufs.Request request){
        return request.getId();
    }

    public static Protobufs.Request createLoginRequest(Referee referee){
        Protobufs.Referee refereeProtobufs = Protobufs.Referee.newBuilder()
                .setEmail(referee.getEmail()).setPassword(referee.getPassword())
                .build();
        Protobufs.Request request = Protobufs.Request.newBuilder()
                .setType(Protobufs.Request.Type.LOGIN)
                .setReferee(refereeProtobufs)
                .build();
        return request;
    }

    public static Protobufs.Request createAddTestRequest(Test test){
        Protobufs.Test testProtobufs = Protobufs.Test.newBuilder()
                .setIdParticipant(test.getIdParticipant())
                .setIdReferee(test.getIdReferee())
                .setPoints(test.getPoints())
                .setDate(test.getDate())
                .build();
        Protobufs.Request request = Protobufs.Request.newBuilder()
                .setType(Protobufs.Request.Type.ADD_TEST)
                .setTest(testProtobufs)
                .build();
        return request;
    }

    public static Protobufs.Request createLogoutRequest(Referee referee){
        Protobufs.Referee refereeProtobufs = Protobufs.Referee.newBuilder()
                .setId(referee.getId())
                .setName(referee.getName())
                .setEmail(referee.getEmail())
                .setPassword(referee.getPassword())
                .build();
        Protobufs.Request request = Protobufs.Request.newBuilder()
                .setType(Protobufs.Request.Type.LOGOUT)
                .setReferee(refereeProtobufs)
                .build();
        return request;
    }

    public static Protobufs.Request createGetAllParticipantsRequest(){
        Protobufs.Request request = Protobufs.Request.newBuilder()
                .setType(Protobufs.Request.Type.GET_ALL_PARTICIPANTS)
                .build();
        return request;
    }


    public static Protobufs.Request createGetAllActiveParticipantsRequest(int id){
        Protobufs.Request request = Protobufs.Request.newBuilder()
                .setType(Protobufs.Request.Type.GET_ALL_ACTIVE_PARTICIPANTS)
                .setId(id)
                .build();
        return request;
    }

    public static Protobufs.Response createOkResponse(){
        Protobufs.Response response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.OK)
                .build();
        return response;
    }

    public static Protobufs.Response createOkResponse(Referee referee){
        Protobufs.Referee refereeProtobufs = Protobufs.Referee.newBuilder()
                .setId(referee.getId())
                .setName(referee.getName())
                .setEmail(referee.getEmail())
                .setPassword(referee.getPassword())
                .build();
        Protobufs.Response response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.OK)
                .setReferee(refereeProtobufs)
                .build();
        return response;
    }

    public static Protobufs.Response createOkResponse(Test test){
        Protobufs.Test testProtobufs = Protobufs.Test.newBuilder()
                .setId(test.getId())
                .setIdParticipant(test.getIdParticipant())
                .setIdReferee(test.getIdReferee())
                .setPoints(test.getPoints())
                .setDate(test.getDate())
                .build();
        Protobufs.Response response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.OK)
                .setTest(testProtobufs)
                .build();
        return response;
    }

    public static Protobufs.Response createErrorResponse(String message){
        Protobufs.Response response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.ERROR)
                .setError(message)
                .build();
        return response;
    }

    public static Protobufs.Response createGetAllParticipantsResponse(ArrayList<Participant> participants){
        Protobufs.Response.Builder response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.GET_ALL_PARTICIPANTS);
        for(Participant participant : participants){
            Protobufs.Participant participant1 = Protobufs.Participant.newBuilder()
                    .setId(participant.getId())
                    .setName(participant.getName())
                    .setTotalPoints(participant.getTotalPoints())
                    .build();
            response.addParticipants(participant1);
        }
        return response.build();
    }

    public static Protobufs.Response createGetAllActiveParticipantsResponse(ArrayList<Participant> activeParticipants){
        Protobufs.Response.Builder response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.GET_ALL_ACTIVE_PARTICIPANTS);
        for(Participant participant : activeParticipants){
            Protobufs.Participant participant1 = Protobufs.Participant.newBuilder()
                    .setId(participant.getId())
                    .setName(participant.getName())
                    .setTotalPoints(participant.getTotalPoints())
                    .build();
            response.addParticipants(participant1);
        }
        return response.build();
    }

    public static Protobufs.Response createRefreshAllResponse(Test test){
        Protobufs.Test test1 = Protobufs.Test.newBuilder()
                .setId(test.getId())
                .setIdParticipant(test.getIdParticipant())
                .setIdReferee(test.getIdReferee())
                .setPoints(test.getPoints())
                .setDate(test.getDate())
                .build();
        Protobufs.Response response = Protobufs.Response.newBuilder()
                .setType(Protobufs.Response.Type.TEST_ADDED)
                .setTest(test1)
                .build();
        return response;
    }
}
